src/offset/local/mod.rs

```diff
@@ -186,6 +190,102 @@ impl TimeZone for Local {
     }
 }
 
+#[cfg(any(unix, windows))]
+pub(crate) struct TzInfo {
+    std_offset: FixedOffset,
+    dst_offset: FixedOffset,
+    std_transition: Option<NaiveDateTime>,
+    dst_transition: Option<NaiveDateTime>,
+}
+
+#[cfg(any(unix, windows))]
+impl TzInfo {
+    fn lookup_with_dst_transitions(&self, dt: &NaiveDateTime) -> LocalResult<FixedOffset> {
```

Comment:
It took me hours to convince myself the logic in this function was correct in the Unix version, so I simplified it a bit here.

If the first transition is to DST, the year starts in STD, transitions to DST, and ends with STD.
If the first transition is to STD, the opposite.

For every transition we take in local time the earliest clock value, and latest clock value (`transition_min` and `transition_max`).
- If the transition crates a gap, all times exclusive do not exist `(*_transition_min..*_transition_max)`.
- If the transition crates an overlap, all times inclusive do not exist `[*_transition_min..*_transition_max]`.
- The time in between the transitions should have the remaining comparison operator (i.e. `>` if the transition uses `<=`).

The result to return in `LocalResult::Single` is straightforward, `dst_offset` or `std_offset` as appropriate.

The correct order in `LocalResult::Ambiguous` is the offset right before the transition, then the offset right after.

Hope that helps for the review.

Reply:
Added a test :+1:

Reply:
Can you lay your explanation here down in a comment?

If this is created to replace existing logic for the Unix time zone logic, I think we should switch the Unix implementation in the same commit so that it's easier for me to compare the new code with the old code.